src/popup/components/TransactionDetailsBase.vue

```diff
@@ -0,0 +1,473 @@
+<template>
+  <div class="transaction-details-base">
+    <div
+      v-if="!isDexAllowance || isErrorTransaction"
+      class="header"
+    >
+      <TransactionErrorStatus
+        v-if="isErrorTransaction"
+        :return-type="transaction.tx.returnType"
+      />
+      <TransactionTokens
+        :ext-tokens="noneAeCoin"
+        :is-rounded="!!noneAeCoin"
+        :transaction="transaction"
+        :direction="direction"
+        :is-allowance="isDexAllowance"
+        :error="isErrorTransaction"
+        :class="{ reverse: isPool }"
+        icon-size="md"
+      />
+    </div>
+    <div class="content">
+      <TransactionOverview :transaction="transaction" />
+      <div class="explorer">
+        <LinkButton :to="explorerUrl">
+          {{ $t('pages.transactionDetails.explorer') }}
+          <ExternalLink />
+        </LinkButton>
+      </div>
+      <div class="data-grid">
+        <template v-if="isSwap && !isErrorTransaction">
+          <SwapRates :transaction="transaction" />
+          <SwapRoute :transaction="transaction" />
+        </template>
+        <DetailsItem
+          v-if="isErrorTransaction"
+          :label="$t('pages.transactionDetails.reason')"
+          :value="transaction.tx.return"
+          class="reason"
+          data-cy="reason"
+        />
+        <TransactionDetailsPoolTokens
```

Comment:
Should base component include protocol specific components?

Reply:
I propose to deal with it in a separate task.